plugins/nordic-hid/nordic-hid.quirk

```diff
@@ -1,19 +1,5 @@
-# Nordic Semiconductor ASA Mouse nRF52 Desktop
-[HIDRAW\VEN_1915&DEV_52DE]
-Plugin = nordic_hid
-GType = FuNordicHidCfgChannel
-#NordicHidBootloader = B0
-#NordicHidBootloader = MCUBOOT
-NordicHidGeneration = default
-
-# Nordic Semiconductor ASA Keyboard nRF52 Desktop
-[HIDRAW\VEN_1915&DEV_52DD]
-Plugin = nordic_hid
-GType = FuNordicHidCfgChannel
-NordicHidGeneration = default
-
-# Nordic Semiconductor ASA Dongle nRF52 Desktop
-[HIDRAW\VEN_1915&DEV_52DC]
+# Nordic Semiconductor ASA HID Devices
+[HIDRAW\VEN_1915]
```

Comment:
you want to probe every device past and future as a CFG channel? This seems a bit risky.

Reply:
That would affect only HID devices with Nordic's Vendor ID, right?
As the first operation during `setup` we simply exchange a HID feature report with predefined report ID and data. We perform HID feature report set and get sequence using `fu_udev_device_ioctl` (and underlying `ioctl`). If a device does not support the HID feature report ID used by the configuration channel, we observe `ioctl` error - in that case we assume that device does not support configuration channel and error is not propagated further eventually resulting in ignoring the device. Unless I missed something, the main risk here is the case of Nordic's HID device that supports the same HID feature report ID and uses it for other purposes. The spurious HID feature report operation may accidentally trigger an operation on device (an unexpected report data would be rejected by the firmware in most cases anyway I think).

On the other hand, using this try&check approach allows us easily handle new devices (no need to update plugin's quirk - new HID devices related to nRF Desktop reference design are supported out-of-the-box).
Also we do not know PIDs of HID peripherals connected via dongle (old version of firmware used not to support reporting them over configuration channel at all). That leads to problems with applying quirks (they need to be applied before the `setup` is done). Apart from this, the fact that PIDs of dongle connected peripherals are not known in advance may lead to inconsistent behavior (e.g. a device may be handled in LVFS when when connected via dongle but it would not work when connected directly - case when dongle's PID is part of quirk, but device's PID is not).

Reply:
I was thinking about a solution for this for a while, but nothing better comes to my mind. Do you have a suggestion, how to handle this use-case, @hughsie ?

Reply:
Does https://github.com/fwupd/fwupd/blob/main/docs/ds20.md help?

Reply:
nRF Desktop devices may be connected via USB, Bluetooth LE or through an nRF Desktop dongle. Using the `BOS DS20` requires to connect the device directly over USB which may lead to inconsistent behavior that depends on selected connection type (generally not intended in our HID use-case). Configuration channel uses HID feature reports to communicate with the device in very similar way no matter how it's connected to the host, using transport-specific features may not work well with this assumption.